### PR TITLE
feat: exact match file basename filter

### PR DESCRIPTION
**Pull request by poyoho**
fix: #2046

### CODE REVIEW

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Why is it named extra?

**Comment by poyoho** [comment]
Is mean extra match the file basename? @sheremet-va 😁

**Comment by sheremet-va** [comment]
> Is mean extra match the file basename? @sheremet-va 😁

Hm? Issue mentioned the name `--exact`, which is much more descriptive than `--extra`. I would even consider `--exact-filter`

File: packages/vitest/src/node/core.ts [hunk 1]
```diff
@@ -22,6 +22,13 @@ import { VitestCache } from './cache'
 
 const WATCHER_DEBOUNCE = 100
 
+function isMatchTest(a: string, b: string, extra: boolean): boolean {
```
Comment (dammy001) [inline packages/vitest/src/node/core.ts:25]: i think we should move this method to utils directory

**Comment by dammy001** [comment]
We should document this after

File: docs/guide/cli.md [hunk 2]
```diff
@@ -79,6 +79,7 @@ vitest related /src/index.ts /src/hello-world.js
 | `--inspect` | Enables Node.js inspector |
 | `--inspect-brk` | Enables Node.js inspector with break |
 | `-h, --help` | Display available CLI options |
+| `--extra-filter` | extra match file basename to test |
```
Comment (sheremet-va) [inline docs/guide/cli.md:82]: Why extra? "extra" implies "more", here we are limiting it. It should be "exact"

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: docs/guide/cli.md (hunk 2 above)
Comment (poyoho) [reply docs/guide/cli.md:82]: oh! sorry

File: docs/guide/cli.md [hunk 3]
```diff
@@ -79,6 +79,7 @@ vitest related /src/index.ts /src/hello-world.js
 | `--inspect` | Enables Node.js inspector |
 | `--inspect-brk` | Enables Node.js inspector with break |
 | `-h, --help` | Display available CLI options |
+| `--exact-filter` | exact match file basename to test |
```
Comment (sheremet-va) [inline docs/guide/cli.md:82]:
```suggestion
| `--exact-filter` | Will only run tests that match the filter exactly. You need to pass down full paths, including root |
```

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vitest/src/node/cli-api.ts [hunk 4]
```diff
@@ -13,6 +13,12 @@ export interface CliOptions extends UserConfig {
    * Override the watch mode
    */
   run?: boolean
+  /**
+   * extra match file to test
```
Comment (sheremet-va) [inline packages/vitest/src/node/cli-api.ts:17]:
```suggestion
   * Will only run tests that match the filter exactly. You need to pass down full paths, including root
```

**Review by sheremet-va — APPROVED** [review]

**Comment by sheremet-va** [comment]
Now thinking about it, I don't think we need this feature. Users can just pass down full path to filter, and it will work.

**Comment by poyoho** [comment]
> Now thinking about it, I don't think we need this feature. Users can just pass down full path to filter, and it will work.

like `base.spec.ts`

**Pull request closed without merging**